include/deal.II/base/symmetric_tensor.h

```diff
@@ -528,11 +528,11 @@ namespace internal
  * @ingroup geomprimitives
  * @author Wolfgang Bangerth, 2005
  */
-template <int rank, int dim, typename Number>
+template <int rank_, int dim, typename Number>
```

Comment:
I'm not a fan of this change because it uglifies a large number of symbols. How about leaving it at `rank` and renaming the static member variable to `tensor_rank`? It's likely that the latter has far fewer occurrences than the former, so the slightly longer name might be acceptable.

Reply:
It would also vastly shrink the patch and make it easier to see the places where something is actually changed at a functional level :-)